### ISSUE TITLE
Bunch of bugs

### ISSUE THREAD

**Issue by Krevik** [SirPlease/L4D2-Competitive-Rework#206]
Okay so I've been playing using this pack lately, but I encountered so much bugs and problems that I decided to collect them all in one issue.

`#1` Randomly occuring: Randomly closing the server with no error logs (the last log line is just cut in the middle) (no matchmode, same with matchmode loaded)
Crash log `*1` - https://crash.limetech.org/t6d6bxpixbea (Here I use newer source mod version as with this source mod version server seems to be more stable [less crashes and less bugs], no matchmode was loaded during gameplay. The server crashed when new player was joining the server)

`#2 Solved` Randomly occuring: Adrenaline is randomly not working (restarting the map or even the server doesn't help) (only without any matchmode loaded)
`Solution` updating the source mod solves the problem, however this may be not a great solution, cause the pack is supposed to run on certain source mod version.

`#3` Almost always: Medkits and starting items are sometimes missing upon joining the server (restarting the map helps) (so far happens only with no matchmode loaded):
 Might be connected with:
`CreateInstance called on a point_template that has no templates: guns_position1`
 but I am not sure. Look at #196 

`#4` Randomly occuring: VISUAL BUG: Some people don't see one of the survivors going through the map (they see one player as he's all the time on his spawn point, he's normally shooting, but not moving through the map) (no matchmode)

`#5` Randomly occuring: People are sometimes randomly shoot in the air when they jump on CI. One time my friend was shoot into the air, while he jumped on the car (no, the car wasn't moved by tank earlier) (matchmode/no matchmode)

`#6` Randomly occuring: Hitboxes of SI and it's model are getting separated - other teammates see the Infected Player in the other place, while he sees himself in the other. It's not lag compensation, it's more like lag compensation crash or ultra lag, cause the player is like in the second half of the map, while other players are seeing him in the first half.  (no matchmode)

`#7` Randomly occuring: While doing instant kill with charger, SURVIVOR sometimes grab the edge, but charger is still charging over the edge + generally instant kills with charger are sometimes bugged - for e.g. server crash after instant kill. (no matchmode)

`#8` Often occuring: Witch sometimes bug badly and blocks at some edge/corner/etc  (matchmode/no matchmode)

`#9` Often occuring: CI blocks on edges especially while jumping and their model stays in the jumping phase (matchmode/no matchmode)

`#10` Rare occuring: Molotov/Pipe is sometimes getting stuck in CI/SI, and unblocks after about (20) seconds? (no matchmode)

`#11` Rare occuring: Sometimes players cannot use important props - like tank wagon doors at the sacrifice (they're not even highlighted on blue) (no matchmode loaded) 

`#12` Randomly Occuring: randomly kicking players because of
`L 11/29/2020 - 23:08:45: STEAMAUTH: Client Jamal the player received failure code 6`
`L 11/29/2020 - 23:08:45: "Jamal the player<265><STEAM_1:1:198187275><Infected>" disconnected (reason "No Steam logon")`
(matchmode/no matchmode)

`#13 Solved` Often occuring: When player shoot zombies (especially SI, and especially hunter) blood particles are spawned in the wrong place - just in front of the player camera (no matchmode, zonemode)
`Solution`: disable code_patcher.smx

`#14` Rare occuring: When infected like jockey, charger grab the survivor, they both got teleported far far away from the map. Happend on: No mercy, hospital (charger), Dark Carnival, love tunnel (Jockey). No error log on accelelerator. Happens with no matchmode loaded and witch matchmode loaded (Zonemod)

`#15` More like suggestion, not a bug: There's possible to get stuck at c5m3_cemetery. Guide (route): from starting saferoom doors go right, go behind the fence, turn right and go under the pillar.

`#16` Randomly occuring: Tank control bar is sometimes not shown at all (matchmode)

`#17` Randomly occuring: People you can make a vote kick for are mixed-up: you can start kick vote for some people from the other team, while you can't make vote for all the people in your team. (matchmode checked only so far)

--------------------------------Some error log lines (without crash) that I do not know where to pin-------------------------
https://justpaste.it/2yoxj
----------------------------------------------------------------------------------------------------------------------------------------------

These are the ones I remember + most of bugs doesn't produce any error logs. Server is running on `Debian`, 100 tickrate.
I have another (CLEAR, 30tickrate) server on the same hosting and I am not encountering any problems on that 30tickrate server (for a while I was sure it has been hosting-associated problem). 
I was reinstalling pack from this repo for ... I don't know maybe like 5 times? I Was trying to keep it as original as possible (I was editing only server config files and motd) but that doesn't help. I love your pack and was playing on your servers and really would like to have working one with this pack but at the moment I don't really know what to do. Server with this pack is mostly unplayable. I see, other users do not open so much issues, so I guess the problem is on my side? If so, do you have any ideas what am I doing wrong? If I should close this issue, just let me know.
Greetings, Krevik.

**Comment by SirPlease** [comment]
Before I touch on the issues, I do want to point out that this project is for Competitive Servers, thus it doesn't focus on vanilla **at all**.
You've reported a lot of issues that aren't part of the project, nor do you state what issue occured when. (Was a config loaded or not?)

Also, you state that you use Linux, if you use Ubuntu and you're using a newer version than 18.04, you're bound to run into tons of issues.

`#1`: Not related to this pack itself, unless you haven't been keeping up to date (a crash was recently fixed, I highly recommend setting up [accelerator](https://forums.alliedmods.net/showthread.php?t=277703) to help you debug crashes and the like more efficiently.

`#2`: We've talked this in a previous issue, I'm not able to reproduce it on my end. There was a recompile of the plugin suggested in the thread and if that didn't help, then I don't know what would because I looked at the code for it and it should work fine unless you've got something else running alongside it that messes with the adrenaline convar.

`#3`: I haven't been able to reproduce this myself, but I'll take your word for it. This would be low priority considering the pack focuses on loaded matchmodes, not vanilla gameplay.

`#4`: Not related to the pack, this is a L4D2 bug. To fix this, simply type `record demo` in console and it'll be fixed. This happens to players every now and then that just had their files crc checked against the whitelist.

`#5`: Known issue, exists in base L4D2 but occurs far more frequently at 100 tick.

`#6`: That's lag compensation. The client will always see themselves somewhere else than other players see them.  ([Good read](https://developer.valvesoftware.com/wiki/Lag_compensation))

`#7`: Not related to the pack, never seen this on my own or any other servers that use this pack.

`#8`: Not related to the pack per se, the witch on itself is already quite buggy. Lowering nb_update_frequency bugs her as well as common out even further. This is also the reason why Zonemod doesn't use the witch in its config, it doesn't belong in a competitive environment.

`#9`: See above.

`#10`: Throwables are able to get stuck on infected, but this is a Valve thing.

`#11`: Not related to the pack.

`#12`: Not related to the pack, user couldn't authenticate with steam, gets kicked.

`#13`: This is where I know you're using a newer version of Ubuntu, as this occurs consistently on the latest libraries. Not related to the pack.

**Comment by Krevik** [comment]
> Before I touch on the issues, I do want to point out that this project is for Competitive Servers, thus it doesn't focus on vanilla **at all**.
> You've reported a lot of issues that aren't part of the project, nor do you state what issue occured when. (Was a config loaded or not?)
> 
> Also, you state that you use Linux, if you use Ubuntu and you're using a newer version than 18.04, you're bound to run into tons of issues.
> 
> `#1`: Not related to this pack itself, unless you haven't been keeping up to date (a crash was recently fixed, I highly recommend setting up [accelerator](https://forums.alliedmods.net/showthread.php?t=277703) to help you debug crashes and the like more efficiently.
> 
> `#2`: We've talked this in a previous issue, I'm not able to reproduce it on my end. There was a recompile of the plugin suggested in the thread and if that didn't help, then I don't know what would because I looked at the code for it and it should work fine unless you've got something else running alongside it that messes with the adrenaline convar.
> 
> `#3`: I haven't been able to reproduce this myself, but I'll take your word for it. This would be low priority considering the pack focuses on loaded matchmodes, not vanilla gameplay.
> 
> `#4`: Not related to the pack, this is a L4D2 bug. To fix this, simply type `record demo` in console and it'll be fixed. This happens to players every now and then that just had their files crc checked against the whitelist.
> 
> `#5`: Known issue, exists in base L4D2 but occurs far more frequently at 100 tick.
> 
> `#6`: That's lag compensation. The client will always see themselves somewhere else than other players see them. ([Good read](https://developer.valvesoftware.com/wiki/Lag_compensation))
> 
> `#7`: Not related to the pack, never seen this on my own or any other servers that use this pack.
> 
> `#8`: Not related to the pack per se, the witch on itself is already quite buggy. Lowering nb_update_frequency bugs her as well as common out even further. This is also the reason why Zonemod doesn't use the witch in its config, it doesn't belong in a competitive environment.
> 
> `#9`: See above.
> 
> `#10`: Throwables are able to get stuck on infected, but this is a Valve thing.
> 
> `#11`: Not related to the pack.
> 
> `#12`: Not related to the pack, user couldn't authenticate with steam, gets kicked.
> 
> `#13`: This is where I know you're using a newer version of Ubuntu, as this occurs consistently on the latest libraries. Not related to the pack.


Thanks for fast answer! About #6 I know what's lag compensation and it's not the thing I am talking about - player is like completely in another place far far away from the location that other players see.

Anyway, thanks, gonna contact my hosting about exact version and try to do something about it.

**Comment by Krevik** [comment]
Okay so now I know my servers are running on "Debian 10". I guess it can be source of all the problems ?

**Comment by SirPlease** [comment]
I used to run stuff on older Debian, but I haven't worked on Debian for a while.
with issue `#13` it's likely related to the newer libraries that updated linux builds run, which is why we've stuck to Ubuntu 16.04~18.04 as those are known not to have this issue.

**Comment by ProjectSky** [comment]
#13 Same issue on ubuntu20.04,  try disable code_patcher.smx

**Comment by Krevik** [comment]
> `#13` Same issue on ubuntu20.04, try disable code_patcher.smx

That seems to work, thanks a lot!

Updated the post (made it more clear), added crash log and `#14`

**Comment by ProjectSky** [comment]
`#15` Same issue (server system version is too new?)
